src/current/v25.1/understand-hotspots.md

```diff
@@ -0,0 +1,313 @@
+
+It is especially useful when correlating pressure with activity, for example, the read pressure on Node 1 is coming from the key `/User/10`. Pressure is a useful term to describe resource limits, because it captures the ideas of [contention]({% link {{ page.version.version }}/performance-best-practices-overview.md %}#transaction-contention), utilization, and throughput - all of which can be the source of a hotspot - under a single definition.
+
+### Index Tail
+
+**Synonyms:** Index Maximum Present Key
+
+An _index tail_ is the largest existing key of an index, lexigraphically. Note that this is different from the index (such as `/Table/1/1`) or index spans (such as `[/Table/1/1, /Table/1/2)`), since these are not valid row keys. An index tail instead refers to the last real key which appears in the index. For example, the head and tail of the index `/Table/1/1` are: `[/Table/1/1/aardvark, /Table/1/1/zultan]` 
+
+## Patterns
+
+This section goes into detail about workload patterns which result in hotspots.
+
+### Index Hotspot
+
+**Synonyms**: hot index, golden keyspace hotspot, monotonically increasing index, a running tail, a moving tail
+
+An _index hotspot_ is a hotspot on an [index]({% link {{ page.version.version }}/indexes.md %}) where the key for writes is continually increasing. This is common with indexing by an increasing column. For example, the column may be of data type [`SERIAL`]({% link {{ page.version.version }}/serial.md %}), [`TIMESTAMP`]({% link {{ page.version.version }}/timestamp.md %}), or [`AUTO_INCREMENT`]({% link {{ page.version.version }}/serial.md %}#auto-incrementing-is-not-always-sequential)). However, an index hotspot is not always determined by the data type. If sequential data generated by the application is inserted into an index, a hotspot may occur. Index hotspots limit horizontal scaling as the index acts as a bottleneck.
+
+Consider a table `users` which contains a [primary key]({% link {{ page.version.version }}/primary-key.md %}) `user_id` that is an incrementing integer value. Each new key will be the current maximum key + 1. In this way, all writes appear at the index tail. The following image visualizes writes to the `users` table using an incrementing `INT` primary key. Note how all writes are focused at the tail of the index, represented by the red section in Range 4.
+
+<img src="{{ 'images/v25.1/hotspots-figure-3.png' | relative_url }}" alt="incrementing INT primary key" style="border:1px solid #eee;max-width:100%" />
+
+If you are able to control for the performance degradation that occurs in Range 4 in this case, you do not change the fact that the system is now limited by how many writes a single range can execute. This then, in theory, limits CockroachDB to the performance of a single node, which is, by nature, antithetical to the goals of a distributed database.
```

Comment:
I found the first sentence a little hard to parse, does it mean "even you do x, you won't be able to do y"?
```suggestion
Even if you are able to control for the performance degradation that occurs in Range 4, it isn't possible to change that the system is now limited by how many writes a single range can execute. As a result, CockroachDB could be limited to the performance of a single node.
```

I think the final sentence here could be more direct. I gave a suggestion a shot, but I'm not sure if the meaning is right here.